addons/mail/static/src/messaging/entity/chatter/chatter.js

```diff
@@ -116,6 +128,28 @@ function ChatterFactory({ Entity }) {
             return [['replace', this.activities.filter(activity => activity.state === 'today')]];
         }
 
+        /**
+         * @private
+         */
+        _prepareAttachmentsLoading() {
+            this._isPreparingAttachmentsLoading = true;
+            const loadingDelay = this.env.disableLoadingDelay ? 0 : 400;
```

Comment:
Store the value 0 or 400 directly in the env, it will be more customizable. Name the env key accordingly.

Reply:
wouldn't assume all loading delay are the same, and that's overkill to have one delay entry per delay loading timer. IMO just a boolean is good enough, and value is decided on a case by case.

Reply:
I feel like all delays should be the same for the consistency of the user experience.

One variable per delay is definitely overkill.

I think if we want specific slower or longer delay in some cases, they should just multiply the base delay by a factor.

Reply:
Should indeed avoid having lots of different delays, or UI will look like a Christmas tree, but it's restrictive to only have one delay, like here 400ms is a good fit but with blocking "Please wait" on non-shadow RPC 3seconds is a better fit.

The intent of this env entry is to enable or disable loading delays. This is quite similar to animation, so it's tempting to merge both entry... Anyway, this looks like a toggle, so it should be a boolean IMO.

Reply:
The blocking "please wait" is an exception, and it should do base delay x10 for example. 

This is consistent with bootstrap variables and configuration too.

There are base values, and there are multipliers, so that if you want to speed up or slow down the interface (or increase spacers, or whatever you like), everything stays consistent relatively to the rest thanks to the use of a factor of the base value.

So I'm not suggesting one delay, I'm suggesting one base delay. This does everything a "toggle" would do and a lot more as a bonus.

So in this case, the component will use the base delay (x1), on another component it can do the base delay (x10), etc.

Reply:
Still no consensus if I understand this right :D ?

Reply:
Still prefer boolean for enable/disable of animations/loading, because that's our main use-case and I think that value of delay depends on appreciation of users, hence I prefer hard-coded value here.

But we can postpone this problem for latter and just pick the one you prefer... 😈

Reply:
Don't see any person-related preference here but there is nothing to change if I stick to @alexkuhn 's solution so I'll leave it this way :angel: :smile:

Reply:
The current solution has an inconsistency though, by setting timeout 0 we still have a delay of "one timeout", so it is technically not a disable, so the variable is incorrectly named or not doing its job.

If we go back it should be all the way (no timeout if disable), but I of course prefer the new solution, but also then all the way (variable timeout with appropriate name and value).

Due to this difference between the two I would even argue that both variables could have their places at the same time! But we don't actually need those features...

Our main and only use case is to have the delay feature working, it's always supposed to be there because we never want a flicker.
So we don't actually want to test a case where there is no timeout, because in reality there will always be one. Since we don't want tests to be too slow, we allow to customize the base delay value, and we set it as low as possible in the tests (= 0), and we get as bonus that we value can be customized easily.

So please update for this solution.

This will also allow us to test it in practice for several runbot until we have a final solution (potentially issue, for example in case sometimes the timeout comes before the afterNextRender for whatever reason).

Reply:
The information provided by component `Chatter` without loading is correct, so it's definitely essentially working code without loading.
The real-use case has loading delays, so I agree it makes sense to test it with loading, notably because there are complex stuffs regarding loading.

But we take a step back, we see that inclusion of loading delays make code more complex (because concurrency at perceptible time frame), in addition to making tests slower and themselves should carefully handle concurrency.

I think animations and loading delays are primarily there just to make things nicer, but this creates lots of issues by themselves.
Logically, we don't need them, and it's much easier to understand, maintain, and test code without them.

All the code is much easier to understand without them, so we should avoid having them when its unnecessary.
The majority of our tests should not care for these delays, and it's already complex enough without them so adding them just add too much burden on these tests.

In my mind, animation and loading delays can be seen like an "with_animations/with_loading" plugin (or 2 distinct plugins? I dunno, that's why it would also make sense to have 2 distinct env keys). Animation and loading delays should ideally act like CSS, which means your website should still working without CSS (but it's ugly and a pain to use... if some stuffs don't work, then you have a much bigger issue that your website CSS is essential...).

Unfortunately, animation and loading delays cannot be as lightly coupled as CSS at the moment. There may be some programming paradigms somewhere that can be aware of this issue and design animation and loading delays in a way that make them non-optional, doesn't break the application if they do not work as expected (or none at all), etc...

So back to my point: IMO no animations and loading delays must still make everything work essentially, and by essentially I mean all our current tests should ideally work and have no `afterNextRender()` or `nextMicroTaskTick()` or `afterNextAnimationFrameOfNextAnimationFrameRequested()`, etc. Delays that are there for prettiness, the ones we are talking about, should be tests on their own and be fully optional.

Reply:
![](https://media1.tenor.com/images/d051ac60e7fa67c9a910c41ac1d75149/tenor.gif)

Reply:
I agree with everything you just said, but as long as this animation code has to be coupled with the normal code, it has to be tested together too. Testing actual features and being able to detect bugs (especially in complex timeout scenario) is more important than having theoretically clean tests, when both are not reconcilable.

I (already) agreed to reconsider this position when we have those possibilities actually in place, but I'm afraid it's not going to happen any time soon.

I also really would like to get rid of all of those magical wait helpers in tests, and I think we made a decent job at reducing them already.

Reply:
So... we merge this like that and we try to find a consensus on that later ?

Reply:
Yes you can merge if everything else is ok.